lib/tetris.js

```diff
@@ -205,30 +240,43 @@ function runFnAtPiece (originalMatrix, piece, pos, rotation, runFunction) {
   }
   return matrix
 }
-
 function insertPiece (originalMatrix, piece, pos, rotation) {
   return changePiece(originalMatrix, piece, pos, pieceId, rotation)
 }
-
 function removePiece (originalMatrix, piece, pos, rotation) {
   return changePiece(originalMatrix, piece, pos, () => 0, rotation)
 }
-
-function copy (object) {
-  return JSON.parse(JSON.stringify(object))
-}
-
-function move (originalPlayer, newPos) {
-  let player = copy(originalPlayer)
-  player.piecePos = newPos
-  player.matrix = removePiece(player.matrix, player.piece, originalPlayer.piecePos, player.pieceRotation)
-  player.matrix = insertPiece(player.matrix, player.piece, player.piecePos, player.pieceRotation)
-  return player
-}
-
 function newPos (pos, direction) {
   return {x: pos.x + xDiff(direction), y: pos.y + yDiff(direction)}
 
   function xDiff (direction) { return direction === 'D' ? 1 : 0 }
   function yDiff (direction) { return (direction === 'R' ? 1 : 0) - (direction === 'L' ? 1 : 0) }
 }
+function randomPiece () {
+  return ['I', 'J', 'L', 'O', 'S', 'T', 'Z'][Math.floor(Math.random() * 7)]
```

Comment:
This function is not pure because of the Math.random call. I'd like to keep only pure functions in this file, mainly by testing reasons. I agree that this function fit here, is a internal call, but I'm not sure how we can solve this conflict. Maybe receive the random as parameter, but that would come through tick and initialState APIs, and it doesn't look too good either. Maybe it would help to think foward about showing the next piece and see if some solution appear.